spec/features/user_feature_spec.rb

```diff
@@ -0,0 +1,42 @@
+require 'rails_helper'
+
+feature "User can sign in and out" do
+  before(:each) do
+    visit('/')
+  end
+  context "user not signed in and on the homepage" do
+
+    it "should see a 'sign in' link and a 'sign up' link" do
+      expect(page).to have_link('Sign in')
+      expect(page).to have_link('Sign up')
+    end
+
+    it "should not see 'sign out' link" do
+      expect(page).not_to have_link('Sign out')
+    end
+
+    it "should not let pictures be posted" do
+      click_link 'New Post'
+      expect(page).to have_content("You need to sign in or sign up before continuing.")
```

Comment:
Metrics/LineLength: Line is too long. [87/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)